events/events_test.go

```diff
@@ -0,0 +1,114 @@
+package events
+
+import (
+	"context"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/reddit/baseplate.go/mqsend"
+
+	"github.com/apache/thrift/lib/go/thrift"
+)
+
+type mockTStruct struct{}
+
+func (mockTStruct) Read(_ thrift.TProtocol) error {
+	return nil
+}
+
+func (mockTStruct) Write(p thrift.TProtocol) error {
+	if err := p.WriteMessageBegin("mock", thrift.CALL, 0); err != nil {
+		return err
+	}
+	return p.WriteMessageEnd()
+}
+
+func TestV2Put(t *testing.T) {
+	const queueSize = 100
+	const doubleTime = DefaultMaxPutTimeout * 2
+	const tripleTime = DefaultMaxPutTimeout * 3
+
+	var v2 *Queue
+	var queue *mqsend.MockMessageQueue
+	t.Run(
```

Comment:
🔕 why run setup code like this under `t.Run`?

Reply:
Subtests are good because:

1. When a test fails, it shows which subtest failed, so you automatically gets more context info about the failure
2. It limits the scope of `t.Fatal`.

You can think about it as a way to connect a few tests in sequence, and allow state sharing between them.

Reply:
But actually it makes less sense in this particular case so I changed it back to just a single test.